partner_contact_sale_info_propagation/models/res_partner.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+from odoo.tools import config
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def _check_propagation_allowed(self):
+        return bool(
+            not config["test_enable"]
+            or (config["test_enable"] and self.env.context.get("test_propagation"))
+        )
+
+    def write(self, vals):
+        """Propagate Salesperson and Sales Channel change in the partner to the
+        child contacts."""
+        if not self._check_propagation_allowed():
+            return super().write(vals)
+        for record in self:
+            if "user_id" in vals:
+                childs = record.mapped("child_ids").filtered(
+                    lambda r, record=record: not r.user_id
+                    or r.user_id == record.user_id
+                )
+                if childs:
+                    childs.write({"user_id": vals["user_id"]})
+            if "team_id" in vals:
+                childs = record.mapped("child_ids").filtered(
+                    lambda r, record=record: not r.team_id
+                    or r.team_id == record.team_id
+                )
+                if childs:
+                    childs.write({"team_id": vals["team_id"]})
+        return super().write(vals)
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        if not self._check_propagation_allowed():
+            return super().create(vals_list)
+        for vals in vals_list:
+            if "parent_id" in vals:
+                if "user_id" not in vals:
+                    vals.update(user_id=self.browse(vals["parent_id"]).user_id.id)
+                if "team_id" not in vals:
+                    vals.update(team_id=self.browse(vals["parent_id"]).team_id.id)
+        return super().create(vals_list)
+
+    @api.onchange("parent_id")
+    def onchange_parent_id(self):
```

Comment:
IMHO, this should be changed to use the compute method instead.

See: https://github.com/odoo/odoo/blob/17.0/odoo/addons/base/models/res_partner.py#L370

Reply:
You're right, its changed now, please if possible check the changes.
Thanks for the review.

Reply:
Yeah you're right, ill work on it.